hedera-node/docs/design/services/smart-contract-service/scheduled-create-token-system-contract.md

```diff
@@ -0,0 +1,91 @@
+| `0xc42a9a17`           | `scheduleUpdateTokenInfo(address token, HederaToken memory tokenInfo)`                                                                                                              | `function scheduleUpdateTokenInfo(address,(string,string,address,string,bool,int64,bool,(uint256,(bool,address,bytes,bytes,address))[],(int64,address,int64))) returns (int64 responseCode, address scheduleAddress)`                                                                                                    | ScheduleCreate  | Schedule token update                                                 |
+
+### System Contract Module
+
+- `ScheduledTokenCreateTranslator` - This class will be responsible for handling the `scheduleCreateFungibleToken`, `scheduleCreateFungibleTokenWithCustomFees`, `scheduleCreateNonFungibleToken`, `scheduleCreateNonFungibleTokenWithCustomFees` and selectors and dispatching them to the Schedule Service.
+- `ScheduledTokenCreateDecoder` - This class provides methods and constants for decoding the given `HssCallAttempt` into a `ScheduleCreateTransactionBody`.
+- `GetScheduledInfoTranslator` - This class will be responsible for handling the `getScheduledFungibleTokenCreateTransaction`, `getScheduledNonFungibleTokenCreateTransaction` and `getScheduledTokenAddress` selectors and dispatching them to the Schedule Service.
+- `GetScheduledTokenInfoCall` - This class provides methods and constants for decoding the `ScheduleGetInfoResponse` into a `PricedResult`.
+- `GetScheduledNonFungibleTokenInfoCall` - This class provides methods and constants for decoding the `ScheduleGetInfoResponse` into a `PricedResult`.
+- `GetScheduledTokenAddressCall` - This class provides methods and constants for decoding the `ScheduleGetInfoResponse` into a `PricedResult`.
+- `ScheduledUpdateTokenInfoTranslator` - This class will be responsible for handling the `scheduleUpdateTokenInfo` selector and dispatching it to the Schedule Service.
+- `ScheduledUpdateTokenInfoDecoder` - This class provides methods and constants for decoding the given `HssCallAttempt` into a `ScheduleCreateTransactionBody`.
+
+### Gas Costs
+
+The gas costs will consist of the price for the call to the Schedule Service plus the pricing for the transaction that would be executed (for creation transactions this would include sending value with the call to cover the fees) and the intrinsic gas cost plus mark-up.
+
+### Feature Flags
+
+In order to gate the newly introduced system contract calls, we will introduce the following feature flags:
+- `systemContract.scheduleService.systemContracts.enabled` - to enable the `scheduleCreateFungibleToken`, `scheduleCreateFungibleTokenWithCustomFees`, `scheduleCreateNonFungibleToken`, `scheduleCreateNonFungibleTokenWithCustomFees` and `scheduleUpdateTokenInfo` functions.
```

Comment:
There's already a `systemContract.scheduleService.enabled` to enable calls to the schedule service system contract.  Should there be another one for the state changing functions here?

Reply:
Changed the naming here.